doc/tunnel/SAI-Proposal-EVPN-Multihoming.md

```diff
@@ -0,0 +1,615 @@
+
+
+  - nh_grp_oid_1 of type SAI_OBJECT_TYPE_NEXT_HOP_GROUP for the remote VTEP group (VTEP1-4)
+
+```
+    sai_attribute_t nhg_attr;
+    vector<sai_attribute_t> nhg_attrs;
+
+    nhg_attr.id = SAI_NEXT_HOP_GROUP_ATTR_TYPE;
+    nhg_attr.value.s32 = SAI_NEXT_HOP_GROUP_TYPE_BRIDGE_PORT;
+    nhg_attrs.push_back(nhg_attr);
+
+    sai_object_id_t next_hop_group_id;
+    sai_status_t status = sai_next_hop_group_api->create_next_hop_group(&nh_grp_oid_1,
+            gSwitchId,
+            (uint32_t)nhg_attrs.size(),
+            nhg_attrs.data());
+
+```
+  - nh_grp_oid_1 has members nh_grp_mbr_oid_1 to 4
+
+```
+    vector<sai_attribute_t> nhgm_attrs;
+    sai_attribute_t nhgm_attr;
+
+    nhgm_attr.id = SAI_NEXT_HOP_GROUP_MEMBER_ATTR_NEXT_HOP_GROUP_ID;
+    nhgm_attr.value.oid = nh_grp_oid_1;
+    nhgm_attrs.push_back(nhgm_attr);
+
+    nhgm_attr.id = SAI_NEXT_HOP_GROUP_MEMBER_ATTR_NEXT_HOP_ID;
+    nhgm_attr.value.oid = nh_oid_1; 
+    nhgm_attrs.push_back(nhgm_attr);
+
+    status = sai_next_hop_group_api->create_next_hop_group_member(&nhgmbr_id_1, gSwitchId,
+            (uint32_t)nhgm_attrs.size(),
+            nhgm_attrs.data());
+
+    /* create_next_hop_group_member for nh_oid_2 to 4 */
+    ..................
+    ..................
+
+```
+
+  - bp_nh_grp_1_oid of type SAI_OBJECT_TYPE_BRIDGE_PORT with nexthop group pointing to nh_grp_oid_1
+
+```
+      sai_attribute_t attr;
+      vector<sai_attribute_t> attrs;
+
+      attr.id = SAI_BRIDGE_PORT_ATTR_TYPE;
+      attr.value.s32 = SAI_BRIDGE_PORT_TYPE_NEXT_HOP_GROUP;
+      attrs.push_back(attr);
+
+      attr.id = SAI_BRIDGE_PORT_ATTR_BRIDGE_PORT_NEXT_HOP_GROUP_ID;
+      attr.value.oid = nh_grp_oid_1;
+      attrs.push_back(attr);
+
+      attr.id = SAI_BRIDGE_PORT_ATTR_BRIDGE_ID;
+      attr.value.oid = m_default1QBridge;
+      attrs.push_back(attr);
+
+      attr.id = SAI_BRIDGE_PORT_ATTR_ADMIN_STATE;
+      attr.value.booldata = true;
+      attrs.push_back(attr);
+
+      attr.id = SAI_BRIDGE_PORT_ATTR_FDB_LEARNING_MODE;
+      attr.value.s32 = SAI_BRIDGE_PORT_FDB_LEARNING_MODE_DISABLE; 
+      attrs.push_back(attr);
+
+      status = sai_bridge_api->create_bridge_port(&bp_nh_grp_1_oid, gSwitchId, 
+                                                  (uint32_t)attrs.size(), attrs.data());
+```
+
+  - mac_host2 pointing to bp_nh_grp_1_oid
+
+```
+        
+            sai_fdb_entry_t fdb_entry;
+ 
+            fdb_entry.switch_id = gSwitchId;
+            memcpy(fdb_entry.mac_address, mac_host2, sizeof(sai_mac_t));
+            fdb_entry.bv_id = entry->bv_id;
+
+            sai_attribute_t attr;
+            vector<sai_attribute_t> attrs;
+
+            attr.id = SAI_FDB_ENTRY_ATTR_TYPE;
+            attr.value.s32 = SAI_FDB_ENTRY_TYPE_DYNAMIC;
+            attrs.push_back(attr);
+
+            attr.id = SAI_FDB_ENTRY_ATTR_BRIDGE_PORT_ID;
+            attr.value.oid = bp_nh_grp_1_oid;
+            attrs.push_back(attr);
+
+            attr.id = SAI_FDB_ENTRY_ATTR_ALLOW_MAC_MOVE;

```

Comment:
Does this operation have any relation with saihost "SAI_HOSTIF_TRAP_TYPE_STATIC_FDB_MOVE". Default command is set  to "drop". Does application need to configure explicit trap object with action "forward" ?

Reply:
Given that ALLOW_MAC_MOVE is set to true there will be no need to 
trap/drop the packets.

Reply:
> Given that ALLOW_MAC_MOVE is set to true there will be no need to trap/drop the packets.

So the assumption here is that "ALLOW_MAC_MOVE = true" on specific fdb entry overrides the packet command configured on SAI_HOSTIF_TRAP_TYPE_STATIC_FDB_MOVE ?

Reply:
Please check the discussion on the above in the below PR where ALLOC_MAC_MOVE was introduced.

https://github.com/opencomputeproject/SAI/pull/1024